### ISSUE TITLE
Why do we call the asset data metadata?

### ISSUE THREAD

**Issue by sbellem** [bigchaindb/bigchaindb#1092]
See:

* https://github.com/bigchaindb/bigchaindb/blob/master/bigchaindb/common/transaction.py#L491-L492

* https://github.com/bigchaindb/bigchaindb/blob/master/bigchaindb/common/schema/transaction.yaml#L101-L116

And what is the asset data then?

**Comment by libscott** [comment]
Good question

**Comment by sohkai** [comment]
@TimDaub should we change it then? Maybe we can earmark this for next week.

**Comment by TimDaub** [comment]
> Maybe we can earmark this for next week.

Yes, this fits perfectly into the feature freeze thingy next week.

**Comment by ttmc** [comment]
I can understand the thinking. The information stored in BigchainDB isn't necessarily _the asset itself_, it's typically just "metadata" about the asset.

Nevertheless, it's confusing to describe the contents of the "asset" key as metadata when there's another key named "metadata"! (The one is really _metadata about the transaction_.)

Anyway, maybe the description of the contents of the "asset" key could say it's information or data about the asset, rather than asset metadata.

## Alternate crazy idea

* Rename "asset" to "asset_metadata"
* Rename "metadata" to "tx_metadata" (or just leave it alone, since it's already transaction["metadata"])

**Comment by sbellem** [comment]
Ok, maybe we should first agree on a definition of metadata? I was assuming [this one](https://www.merriam-webster.com/dictionary/metadata):

> Metadata is "data [information] that provides information about other data".

On that basis, my question is simply if the user is passing the metadata of the asset, what about the data of the asset, what is it, where is it?

**Comment by ttmc** [comment]
There are three possible cases:

1. The asset itself is stored inside the "asset". For example, maybe it's the text of an entire essay.
2. Metadata about the asset is stored inside the "asset", e.g. the bicycle's serial number, manufacturer, production year, retailer, etc.
3. A mix of 1 and 2.

I think case 2 will be more common, because transactions can only store up to 15kb, so you won't be storing big digital photos or videos in one. But what do I know?

Case 1 _is possible_ and in that case the contents of "asset" are definitely not metadata.

**Comment by libscott** [comment]
Other option: remove asset data and have transaction.metadata as the only field with user definable payload.

**Comment by sbellem** [comment]
So if the data associated with an asset can potentially be either _metadata_, or just _data_, then why not simply call it _data_? Since _metadata_ is a specialized form of _data_ calling it _data_ does not exclude it.

**Comment by r-marques** [comment]
This issue no longer seems valid.
The asset now has a `data` field only in `CREATE` transactions where the asset is definied https://github.com/bigchaindb/bigchaindb/blob/v0.9.5/bigchaindb/common/schema/transaction.yaml#L114

And all transactions still have the `metadata` field.

@sbellem thoughts?

**Comment by sbellem** [comment]
@r-marques indeed, closing the issue.
